pages/api/extension/relevant.ts

```diff
@@ -0,0 +1,90 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import { getReviewData, getFileData, getHunkData } from '../../../utils/db/relevance'
+import { getToken } from 'next-auth/jwt'
+import { getUserEmails } from '../../../utils/db/users'
+
+const getUser = async (req: NextApiRequest, res: NextApiResponse) => {
+  const user = await getToken({req: req})
+  return (user?.email) ? await getUserEmails(user.email) : new Set<string>();
+}
+
+const relevantHandler = async (req: NextApiRequest, res: NextApiResponse) => {
+  // For cors prefetch options request
+  if (req.method == "OPTIONS") {
+    res.setHeader("Access-Control-Allow-Headers", "Access-Control-Allow-Origin, Content-Type, Authorization");
+    res.status(200).send("Ok");
+    return;
+  }
+  // For normal requests
+  console.info("Getting relevant info for ", req.body.repo_name);
+  const user_emails = await getUser(req, res);
+  const { type } = req.query;
+  if (!("repo_provider" in req.body) || 
+    !("repo_owner" in req.body) || 
+    !("repo_name" in req.body)) {
+    res.status(401).send('Invalid request body');
+  }
+  let formattedData;
+  if (type === 'review') {
+      let review_db = await getReviewData(req.body.repo_provider,
```

Comment:
Low LOGAF: we can have a better naming here, `review_data_from_db` ?

Reply:
also, camelCase instead of snake_case)

Reply:
this can also be a `const`.